packages/terra-dev-site/src/webpack/plugin/SitePlugin.js

```diff
@@ -0,0 +1,329 @@
+
+    // Since there can be multiple dev site plugins this config we only want to do once for all of them.
+    SitePlugin.applyOneTimeSetup({
+      compiler,
+      sourceFolder,
+      distributionFolder,
+      basename,
+      isWebpack5,
+    });
+
+    // Get the list of apps excluding this current app.
+    const filteredSites = Object.values(siteRegistry).filter(site => site.path !== this.siteConfig.pathPrefix);
+
+    // Map to what we want to send to site config
+    const otherSites = filteredSites.map((site) => ({
+      path: site.path,
+      title: site.title,
+      url: `${basename}${site.url}`,
+    }));
+
+    // if there is a path prefix we want to update the react router basename to include the prefix.
+    if (this.siteConfig.pathPrefix) {
+      basename = [basename, this.siteConfig.pathPrefix].join('/');
+    }
+
+    let webpackConfig = {
+      entry: {
+        [this.entryKey]: `@cerner/terra-dev-site/lib/webpack/templates/entry.template${this.resourceQuery}`,
+      },
+      module: {
+        rules: [
+          {
+            // This loader generates the entrypoint and sets up the config template path and resource query.
+            resourceQuery: this.resourceQuery,
+            use: [
+              babelLoader,
+              {
+                loader: 'devSiteEntry',
+                options: {
+                  entryPath: this.entry,
+                  siteConfig: this.siteConfig,
+                  sites: otherSites,
+                  basename,
+                  resolveExtensions: compiler.options.resolve.extensions,
+                  isLernaMonoRepo,
+                  contentDirectory: this.contentDirectory,
+                },
+              },
+            ],
+          },
+        ],
+      },
+    };
+
+    // If this plugin is used with webpack 5 we must normalize the webpack config.
+    if (isWebpack5) {
+      webpackConfig = compiler.webpack.config.getNormalizedWebpackOptions(webpackConfig);
+    }
+
+    // ENTRY
+    compiler.options.entry = {
+      ...compiler.options.entry,
+      ...webpackConfig.entry,
+    };
+
+    // MODULE
+    // we know there is a oneOf here because we just added it.
+    compiler.options.module.rules[0].oneOf.unshift(webpackConfig.module.rules[0]);
+
+    // Generate the index.html file for the site.
+    new HtmlWebpackPlugin({
+      title: this.siteConfig.titleConfig.title,
+      filename: this.htmlFileName,
+      template: path.join(__dirname, '..', 'templates', 'index.html'),
+      favicon: this.siteConfig.faviconFilePath,
+      headHtml: [getNewRelicJS()].concat(this.siteConfig.headHtml),
+      excludeChunks: ['rewriteHistory', 'redirect', ...Object.values(filteredSites).map(site => site.entry)],
+    }).apply(compiler);
+
+    // Clean up the singletons after plugins are applied.
+    compiler.hooks.afterPlugins.tap('terra-dev-site-site-plugin', () => {
+      oneTimeSetupComplete = false;
+      siteRegistry = {};
+    });
```

Comment:
This is new, if you run all themes through wdio the singlton wasn't resetting for each new webpack dev server run. this resets the singleton after the plugins are run.